bundledatasrc.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package charm
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/juju/errors"
+	"gopkg.in/yaml.v2"
+)
+
+type fieldPresenseMap map[interface{}]interface{}
```

Comment:
if  fieldName is always a string, why isn't this a map[string]interface{} ?

Reply:
That's a good observation. I will change it to use string keys instead.

Reply:
Actually, now that I thought about it again, I am using this because it's the default map type used by the yaml package (https://github.com/go-yaml/yaml/blob/v2/decode.go#L237). If I change this to use string keys, nested maps will use interface{} keys.